client/components/platform-checkout/save-user/checkout-page-save-user.js

```diff
@@ -14,18 +14,50 @@ import AdditionalInformation from './additional-information';
 import PhoneNumberInput from './phone-number-input';
 import Agreement from './agreement';
 import { getConfig } from 'utils/checkout';
-import './style.scss';
 
 const CheckoutPageSaveUser = () => {
 	const [ isSaveDetailsChecked, setIsSaveDetailsChecked ] = useState( false );
 	// eslint-disable-next-line no-unused-vars
 	const [ phoneNumber, setPhoneNumber ] = useState( '' );
+	const [ isPhoneValid, onPhoneValidationChange ] = useState( null );
 	const isRegisteredUser = usePlatformCheckoutUser();
 	const {
 		isWCPayChosen,
 		isNewPaymentTokenChosen,
 	} = useSelectedPaymentMethod();
 
+	// This runs once to populate the phone number state if already present in the DOM.
+	useEffect( () => {
+		setPhoneNumber(
+			document.getElementById( 'billing_phone' )?.value ?? ''
+		);
+	}, [] );
+
+	useEffect( () => {
+		const formSubmitButton = document.querySelector(
+			'form.woocommerce-checkout button[type="submit"]'
+		);
+
+		if ( ! formSubmitButton ) {
+			return;
+		}
+
+		const updateFormSubmitButton = () => {
+			if ( isPhoneValid ) {
```

Comment:
`isPhoneValid` is `null` initially and it does not change until the user clicks the `Save my information for faster checkouts` checkbox or edits the phone number input field. For the `null` value the `Place Order` button is disabled though I have all the required fields filled up in the original checkout blocks.

![disabled](https://user-images.githubusercontent.com/33387139/175365824-ef24f911-633c-4de6-b96c-430afea6286a.png)

The button gets enabled as soon as I select the checkbox.

Reply:
Can you please let me know how to reproduce this? I can't reproduce it on my end:
<img width="874" alt="Screen Shot 2022-06-28 at 11 48 30" src="https://user-images.githubusercontent.com/12542046/176161160-7e58919e-18f0-4a95-bd60-5dec2060532d.png">

Reply:
To see the `Remember your details?` section, you need to have `WooPay` enabled on your merchant site. 

First, you need to set it up in WCPay server DB. Follow any one of the following steps-
- Run this command `wp wcpay enable_woopay <merchant-blog-id>` in your WCPay Server's container.

OR

- In WCPay Server's `wcpay_account_meta` table, add a meta key `platform_checkout_eligible` for your WCPay account, with a meta value of `1`. To find your WCPay account ID, look at the `wcpay_accounts` table.

Then on your merchant site UI enable **wp-admin > Payments > Settings > Express checkouts > WooPay**

![woopay](https://user-images.githubusercontent.com/33387139/176758067-0fe3467d-bb7c-44c9-8ccb-f1546e4b77d2.png)

Finally on the merchant's checkout page input an email id and select credit card as payment method. The checkbox will appear now. :tada: